App/View Controllers/Posts/PostsPageSettingsViewController.swift

```diff
@@ -127,6 +139,73 @@ final class PostsPageSettingsViewController: ViewController, UIPopoverPresentati
             .receive(on: RunLoop.main)
             .sink { [weak self] in self?.imagesSwitch?.isOn = $0 }
             .store(in: &cancellables)
+
+        $immersionModeEnabled
+            .receive(on: RunLoop.main)
+            .sink { [weak self] in self?.immersionModeSwitch?.isOn = $0 }
+            .store(in: &cancellables)
+
+        // Create immersion mode UI at the end of viewDidLoad
+        setupImmersionModeUI()
+    }
+
+    private func setupImmersionModeUI() {
+        // Create immersion mode label
+        immersionModeLabel = UILabel()
+        immersionModeLabel.text = "Immersion Mode"
+        immersionModeLabel.font = UIFont.preferredFont(forTextStyle: .body)
+        immersionModeLabel.textColor = theme["sheetTextColor"] ?? UIColor.darkText
+        immersionModeLabel.accessibilityTraits = []
+        
+        // Create immersion mode switch
+        immersionModeSwitch = UISwitch()
+        immersionModeSwitch.isOn = immersionModeEnabled // Set initial state
+        immersionModeSwitch.onTintColor = theme["settingsSwitchColor"] // Apply theme color
+        immersionModeSwitch.addTarget(self, action: #selector(toggleImmersionMode(_:)), for: .primaryActionTriggered)
+        immersionModeSwitch.accessibilityLabel = "Hide bars when scrolling"
+        
+        // Create stack view for immersion mode
+        immersionModeStack = UIStackView(arrangedSubviews: [immersionModeLabel, immersionModeSwitch])
+        immersionModeStack.axis = .horizontal
+        immersionModeStack.distribution = .equalSpacing
+        immersionModeStack.spacing = 8
+        immersionModeStack.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Find the main settings stack view and append to it
+        func findMainSettingsStack(in view: UIView) -> UIStackView? {
+            if let stackView = view as? UIStackView,
+               stackView.axis == .vertical,
+               stackView.arrangedSubviews.count >= 4 {
+                return stackView
+            }
+            for subview in view.subviews {
+                if let found = findMainSettingsStack(in: subview) {
+                    return found
+                }
+            }
+            return nil
+        }
```

Comment:
I don't get it, why don't we have an outlet or property with the stack view handy?